### ISSUE TITLE
Livescript enums not working

### ISSUE THREAD

**Issue by ihm-tswow** [tswow/tswow#180]
Looks like livescript enums might not be working, equality checks not working as they should

**Comment by the1whodevs** [comment]
TypeScript: 
```typescript
if (splitMsg.length > 3 || splitMsg.length < 3) InformGearupUsage(player);
else if (!CheckIsClassName(splitMsg[1])) InformGearupUsage(player);
else if (CheckGetRole(splitMsg[2]) == ROLE.Invalid) InformGearupUsage(player);
else if (!CheckClassCanPlayRole(splitMsg[1], CheckGetRole(splitMsg[2]))) InformGearupUsage(player);
else HandleGearUp(msg);
```

Converted C++:
```cpp
if ((splitMsg->get_length() > 3) || (splitMsg->get_length() < 3)) InformGearupUsage(player); 
else if (!CheckIsClassName(const_(splitMsg)[1])) InformGearupUsage(player); 
else if (static_cast<long>(CheckGetRole(const_(splitMsg)[2])) == ROLE::Invalid) InformGearupUsage(player); 
else if (!CheckClassCanPlayRole(const_(splitMsg)[1], CheckGetRole(const_(splitMsg)[2]))) InformGearupUsage(player); 
else HandleGearUp(msg);
```

**Comment by the1whodevs** [comment]
Enum Declaration

TypeScript:
```typescript
enum ROLE { Tank, DPS, Healer, Invalid };

const COMMAND_PREFIX = "-";
const GEARUP_COMMAND = COMMAND_PREFIX + "gearup";
```

Converted C++:
```cpp
enum struct ROLE;

extern TSString COMMAND_PREFIX;
extern TSString GEARUP_COMMAND;

enum struct ROLE 
{
    Tank, DPS, Healer, Invalid
};
```

**Comment by ihm-tswow** [comment]
fix: 9b93d74e9a59eb26de87e6e888fd2fe1e0ba55c7
